### DISCUSSION TITLE
How can I search a subset?

### DISCUSSION THREAD

**Discussion by junshan-bao** [facebookresearch/faiss#3575] [General]
# Summary

```python
index = faiss.index_factory(64, 'IVF1024,PQ16')
index.train(emb)  # emb size 20000000,64
index.add(emb)
```
I want to search a subset indices, like
```python
index.search(query, k=10, sub_index=[1,2,10,11,500,1110, ....])
```
How can I do?

# Platform

Faiss version: 1.7.4

Installed from: pip

Running on:
- [x] CPU
- [ ] GPU

Interface: 
- [ ] C++
- [x] Python

**Comment by guillaumeguy** [comment]
It looks like it's documented here: https://github.com/facebookresearch/faiss/wiki/Setting-search-parameters-for-one-query#searching-in-a-subset-of-elements
